btor/src/boolector_term.cpp

```diff
@@ -519,6 +434,60 @@ bool BoolectorTerm::is_const_array() const
   return ((bn->kind == BTOR_LAMBDA_NODE) && (bn->is_array));
 }
 
+void BoolectorTerm::collect_children()
+{
+  if (btor_node_is_proxy(bn))
+  {
+    bn = btor_node_get_simplified(btor, bn);
+    bn = btor_node_real_addr(bn);
+  }
+  else if (children_cached_)
+  {
+    // children have already been computed on an updated node
+    // can't do this if negated, because arity won't match
+    // boolector will say 0 because it's not considered an operator
+    // but in smt-switch we'd want one child
+    return;
+  }
+
+  if (negated)
+  {
+    // the negated value is the real address stored in bn
+    children.clear();
+    children.push_back(bn);
+    return;
+  }
+
+  children.clear();
+  BtorNode * tmp;
+  // don't expose the parameter node of the lambda -- start at 1 instead of 0
+  size_t start_idx = is_const_array() ? 1 : 0;
+  assert(!is_const_array() || bn->e[0]->kind == BTOR_PARAM_NODE);
+  for (size_t i = start_idx; i < bn->arity; ++i)
+  {
+    tmp = bn->e[i];
+    // TODO: figure out if should we do this?
+    // if (btor_node_is_proxy(tmp))
+    // {
+    //   tmp = btor_node_get_simplified(btor, tmp);
+    // }
```

Comment:
This should not be needed since `bn` would be a proxy then. It should never happen that a non-proxy node has proxy children. You can actually assert that `assert(!btor_node_is_proxy(tmp));`

Reply:
Perfect, just added the assertion. Thanks for the review!